src/openqaoa-core/problems/binpacking.py

```diff
@@ -0,0 +1,314 @@
+        self.weights = weights
+        self.weight_capacity = weight_capacity
+        self.penalty = penalty
+        self.n_items = len(weights)
+        self.method = method
+        self.simplifications = simplifications
+        self.eq_constraints = {}
+        self.ineq_constraints = {}
+        if n_bins is None:
+            self.n_bins = self.n_items
+        else:
+            self.n_bins = n_bins
+
+        self.solution = self.solution_dict()
+        self.cplex_model = self.docplex_model()
+        self.n_vars = self.cplex_model.number_of_binary_variables
+
+    def solution_dict(self):
+        solution = {f"y_{i}": None for i in range(self.n_bins)}
+        for i in range(self.n_items):
+            for j in range(self.n_bins):
+                solution[f"x_{i}_{j}"] = None
+        if self.simplifications:
+            # First simplification: we know the minimum number of bins
+            self.min_bins = int(np.ceil(np.sum(self.weights) / self.weight_capacity))
+            for j in range(self.n_bins):
+                if j < self.min_bins:
+                    solution[f"y_{j}"] = 1
+            solution["x_0_0"] = 1
+            for j in range(1, self.n_bins):
+                solution[f"x_0_{j}"] = 0
+        return solution
+
+    @staticmethod
+    def random_instance(**kwargs):
+        """
+        Creates a random instance of the Bin Packing problem.
+
+        Parameters
+        ----------
+        n_items: int
+            The number of items that must be placed in the bins.
+        seed: int
+            Seed number for choosing the random weights
+        min_weight: int
+            Minimum weight to choose the weights from
+        max_weight: int
+            Maximum weight to choose the random weights from
+        simplification: bool
+            Use the simplification version of the problem
+        method: ["slack", "unbalanced"]
+            Use one of the two methods available to enconde the inequality constraints.
+            The slack method requires extra-qubits.
+
+        Returns
+        -------
+            A random instance of the bin packing problem.
+        """
+        seed = kwargs.get("seed", 1234)
+        np.random.seed(seed)
+        weight_capacity = kwargs.get("weight_capacity", 15)
+        n_items = kwargs.get("n_items", 3)
+        n_bins = n_items
+        min_weight = kwargs.get("min_weight", 1)
+        max_weight = kwargs.get("max_weight", 7)
+        if min_weight >= max_weight:
+            raise ValueError(
+                f"min_weight: {min_weight} must be < max_weight:{max_weight}"
+            )
+        weights = list(np.random.randint(min_weight, max_weight, n_items))
+        simplification = kwargs.get("simplification", True)
+        penalty = kwargs.get("penalty", [])
+        method = kwargs.get("method", "slack")
+        return BinPacking(
+            weights, weight_capacity, penalty, n_bins, simplification, method
+        )
+
+    def docplex_model(self):
```

Comment:
Since this method only requires self. Should this be a property instead? (same for `solution_dict`)